ambry-commons/src/main/java/com.github.ambry.commons/ByteBufferReadableStreamChannel.java

```diff
@@ -65,6 +74,30 @@ public long getSize() {
   }
 
   @Override
+  public void setDigestAlgorithm(String digestAlgorithm)
+      throws NoSuchAlgorithmException {
+    if (digest != null && digest.getAlgorithm().equals(digestAlgorithm)) {
```

Comment:
yes, This is bit confusing. Why can't we have a constraint that setDigest() can be called only once and lets make it clear in the docs about this. Felt like we are complicating things and hence